Source/Scene/Cesium3DTileStyle.js

```diff
@@ -0,0 +1,451 @@
+/*global define*/
+define([
+        '../Core/clone',
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/defineProperties',
+        '../Core/DeveloperError',
+        '../Core/loadJson',
+        '../Core/RequestScheduler',
+        '../ThirdParty/when',
+        './ConditionsExpression',
+        './Expression'
+    ], function(
+        clone,
+        defaultValue,
+        defined,
+        defineProperties,
+        DeveloperError,
+        loadJson,
+        RequestScheduler,
+        when,
+        ConditionsExpression,
+        Expression) {
+    'use strict';
+
+    var DEFAULT_JSON_COLOR_EXPRESSION = 'color("#ffffff")';
+    var DEFAULT_JSON_BOOLEAN_EXPRESSION = true;
+    var DEFAULT_JSON_NUMBER_EXPRESSION = 1.0;
+
+    /**
+     * Evaluates an expression defined using the
+     * {@link https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/master/Styling|3D Tiles Styling language}.
+     *
+     * @alias Cesium3DTileStyle
+     * @constructor
+     *
+     * @param {String|Object} [data] The url of a style or an object defining a style.
+     *
+     * @example
+     * tileset.style = new Cesium.Cesium3DTileStyle({
+     *     color : {
+     *         conditions : [
+     *             ['${Height} >= 100', 'color("purple", 0.5)'],
+     *             ['${Height} >= 50', 'color("red")'],
+     *             ['true', 'color("blue")']
+     *         ]
+     *     },
+     *     show : '${Height} > 0',
+     *     meta : {
+     *         description : '"Building id ${id} has height ${Height}."'
+     *     }
+     * });
+     */
+    function Cesium3DTileStyle(data) {
+        this._style = undefined;
+        this._ready = false;
+        this._readyPromise = when.defer();
+        this._color = undefined;
+        this._show = undefined;
+        this._pointSize = undefined;
+        this._meta = undefined;
+
+        this._colorShaderFunction = undefined;
+        this._showShaderFunction = undefined;
+        this._pointSizeShaderFunction = undefined;
+        this._colorShaderFunctionReady = false;
+        this._showShaderFunctionReady = false;
+        this._pointSizeShaderFunctionReady = false;
+
+        var style = this;
+        if (typeof data === 'string') {
+            RequestScheduler.request(data, loadJson).then(function(styleJson) {
+                setup(style, styleJson);
+                style._readyPromise.resolve(style);
+            }).otherwise(function(error) {
+                style._readyPromise.reject(error);
+            });
+        } else {
+            setup(style, data);
```

Comment:
Is it possible for `setup` to fail or otherwise throw?  If so, the behavior here will be different depending on whether `data` was a url or not.  In the url case, it would reject, in the JSON case, it will throw.  We should make sure the promise is rejected in all cases (probably by always going through a promise here.  I realize some of this will change with the RequestScheduler merge into this branch.